Issue a warning if we are falling back to the old shell environment

bbenne10 opened the pull request:
This should resolve #486.

Mic92 commented on direnvrc:
```diff
@@ -326,6 +326,7 @@ use_flake() {
         # which is probably more useful than nothing.
         # Fallback to use that (which means just leaving profile_rc alone!)
         _nix_direnv_warning "Evaluating current devShell failed. Falling back to previous environment!"
+        export NIX_DIRENV_DID_FALLBACK=1
```
Who is the consumer of this environment variable?

bbenne10 commented:
This is expected to be exported to the calling shell for possible user consumption

-------- Original Message --------
On 4/20/24 7:17 PM, Jörg Thalheim  wrote:

> @Mic92 commented on this pull request.
>
> ---------------------------------------------------------------
>
> In [direnvrc](https://github.com/nix-community/nix-direnv/pull/487#discussion_r1573457874):
>
>> @@ -326,6 +326,7 @@ use_flake() {
>          # which is probably more useful than nothing.
>          # Fallback to use that (which means just leaving profile_rc alone!)
>          _nix_direnv_warning "Evaluating current devShell failed. Falling back to previous environment!"
> +        export NIX_DIRENV_DID_FALLBACK=1
>
> Who is the consumer of this environment variable?
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/nix-community/nix-direnv/pull/487#pullrequestreview-2013189008), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AAFH2EA5YA3AHXLMT4ANHSTY6LZP5AVCNFSM6AAAAABGPX4346VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMJTGE4DSMBQHA).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

Mic92 commented:
Ok. Do you have a concrete use case for it? If so it would be nice to put in the Readme.

bbenne10 commented:
README updated to note the environment variable. Additionally, I've changed the note about `nix-direnv-watch-file` (which is no longer accurate - oops!) and added a note about `hide_env_diff` in (what I believe to be) a more appropriate way than #485.

Mic92 commented:
@mergify queue
